### PR TITLE
refactor(chat): simplify chat execution model

### PR DESCRIPTION
**Pull request by chicoxyzzy**
## Summary

Simplifies the chat execution model by removing the old session-level `runtime_kind` split and replacing it with two clearer concepts:

- `agent_id` owns the chat session (`hecate`, `codex`, `claude_code`, `cursor_agent`, ...).
- `execution_mode` describes each turn/message (`direct_model`, `hecate_task`, `external_agent`).

This lets Hecate Chat keep one session while mixing direct model turns and task-backed tool turns, and keeps External Agent sessions owned by their adapter id without pretending that the whole session is a runtime kind.

## What changed

- Updated chat memory/SQLite stores, API request/response types, renderers, and validation errors to use `agent_id` + per-message `execution_mode`.
- Removed compatibility shims for `runtime_kind`; this is an alpha breaking change by design.
- Updated UI state, Chat view projections, transcript rendering, error diagnostics, E2E fixtures, and tests to use the new model.
- Preserved adapter terminology only where it still describes ACP adapter resources: health, credentials, approvals, and adapter-specific endpoints.
- Updated runtime API docs and RFCs to describe session ownership and turn execution separately.

## Verification

- `go test ./internal/chat ./internal/api`
- `go vet ./internal/chat ./internal/api`
- `cd ui && bun run typecheck`
- `cd ui && bun run test`
- `cd ui && bun run test:e2e chat.spec.ts`

### CODE REVIEW

**Comment by chicoxyzzy** [comment]
Addressed the active review feedback in 7dcd9c40. Summary: empty tools-off Hecate chats now keep direct-model mode, unknown/missing execution_mode no longer implies tools-on, Hecate session workspaces validate on create, adapter-scoped fixtures are back to adapter_id for approvals/grants/credentials/health, the runtime API docs mention the Hecate model requirement, and the agent guidance now makes pre-PR TS/Go verification explicit.\n\nI intentionally did not add SQLite backfill compatibility for old runtime_kind/adapter_id rows because this alpha branch is allowed to make breaking storage changes and we are removing legacy instead of carrying shims.\n\n@copilot please review again.

File: internal/chat/sqlite.go [hunk 1]
```diff
@@ -480,7 +475,7 @@ func (s *SQLiteStore) migrate(ctx context.Context) error {
 		name       string
 		definition string
 	}{
-		{name: "runtime_kind", definition: "TEXT NOT NULL DEFAULT 'external_agent'"},
+		{name: "agent_id", definition: "TEXT NOT NULL DEFAULT 'hecate'"},
```
[the remark this answers is not in the document]
Comment (chicoxyzzy) [reply internal/chat/sqlite.go:478]: Intentional for this alpha-only refactor: we are not carrying SQLite compatibility/backfill for the old chat runtime_kind/adapter_id storage contract. Chat ownership is now agent_id and turn execution is per-message execution_mode; existing alpha chat DBs may be reset/recreated rather than migrated through a shim.

File: internal/chat/sqlite.go [hunk 2]
```diff
@@ -81,8 +80,7 @@ func (s *SQLiteStore) Create(ctx context.Context, session Session) (Session, err
 		),
 		session.ID,
 		session.Title,
-		normalizeRuntimeKind(session),
-		session.AdapterID,
+		normalizeAgentID(session),
```
[the remark this answers is not in the document]
Comment (chicoxyzzy) [reply internal/chat/sqlite.go:83]: Same decision here: we are deliberately removing the legacy hecate_chat_sessions adapter_id/runtime_kind schema instead of keeping transitional writes. This is an alpha breaking change, so fresh SQLite schema/tests cover the new shape rather than preserving old-row compatibility.

File: internal/chat/sqlite.go [hunk 3]
```diff
@@ -56,14 +56,13 @@ func (s *SQLiteStore) Create(ctx context.Context, session Session) (Session, err
 		ctx,
 		fmt.Sprintf(
 			`INSERT INTO %s (
-				id, title, runtime_kind, adapter_id, driver_kind, native_session_id, workspace, workspace_branch,
+				id, title, agent_id, driver_kind, native_session_id, workspace, workspace_branch,
 				status, task_id, latest_run_id, provider, model, capabilities, config_options, rtk_enabled, turns_used, created_at, updated_at
 			 )
-			 VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+			 VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
```
[the remark this answers is not in the document]
Comment (chicoxyzzy) [reply internal/chat/sqlite.go:62]: Checked the current fixture: it no longer writes runtime_kind/adapter_id into hecate_chat_sessions. The remaining adapter_id fields in e2e/approval_reconcile_test.go are approval-record fields, where adapter_id is still canonical.

**Comment by chicoxyzzy** [comment]
Addressed the new review feedback in 16bf0fab and the follow-up format failure in 31e8928d.

What changed:
- External-agent sessions now reject `execution_mode: "direct_model"` before provider dispatch, so Codex/Claude/Cursor transcripts cannot receive direct-model turns.
- Queued-message restore now drops entries without a supported `execution_mode` instead of coercing legacy/missing data into `hecate_task`.
- E2E fixtures now use execution-mode terminology, no longer expose a session-level `execution_mode`, and use the correct `"model"` chat-target storage value.
- The UI fixture was formatted so CI `format:check` passes.

I resolved the fixed/outdated threads. For the SQLite migration comments, I replied on-thread: this alpha refactor intentionally removes the old `runtime_kind`/`adapter_id` chat storage contract instead of adding a compatibility/backfill shim.

Verified locally with:
- `go test ./internal/api ./internal/chat`
- `go vet ./internal/api ./internal/chat`
- `cd ui && bun run format:check`
- `cd ui && bun run typecheck`
- `cd ui && bun run test`

@copilot please review again.

**Comment by chicoxyzzy** [comment]
Follow-up after CI: TypeScript E2E exposed one more refresh-state edge case. Fixed in 9471be7b by making an active Hecate task segment override any stale persisted tools-off preference while the task is queued/running/awaiting approval.

I also dropped the transient Copilot-agent commit because its expectation change was incorrect locally: the onboarding model-selection flow should still show `Tools on` before the first Hecate task turn.

Latest PR checks are green now, including TypeScript tests and TypeScript E2E.

@copilot please review again.

File: internal/api/handler_chat.go [hunk 4]
```diff
@@ -61,29 +62,32 @@ func (h *Handler) HandleCreateChatSession(w http.ResponseWriter, r *http.Request
...
+		workspace = resolved
 	}
 	workspaceBranch := workspaceGitBranch(workspace)
 	title := strings.TrimSpace(req.Title)
 
 	session := chat.Session{
 		ID:              newChatID("chat"),
 		Title:           title,
-		RuntimeKind:     runtimeKind,
+		AgentID:         agentID,
```
[the remark this answers is not in the document]
Comment (chicoxyzzy) [reply internal/api/handler_chat.go:78]: Thanks, checked this path closely. Hecate-owned session creation already defaults an omitted/blank title before persistence (`session.Title = "Hecate Chat"` in `handler_chat.go`). I added `TestHecateAgentChatCreateDefaultsTitle` in `internal/api/handler_chat_hecate_test.go` so this behavior is now explicit and covered. Resolving this thread.

File: ui/src/app/state/coordinators/chat.ts [hunk 5]
```diff
@@ -310,27 +315,27 @@ export function useChatActions(params: UseChatActionsParams): ChatActionsReturn
...
       session_id: sessionID,
       content,
-      runtime_kind: runtimeKind,
+      execution_mode: executionMode,
       provider_filter: providerFilter,
       model,
       workspace: agentWorkspace.trim(),
       system_prompt: systemPrompt,
-      adapter_id: agentAdapterID,
+      agent_id: agentAdapterID,
```
[the remark this answers is not in the document]
Comment (chicoxyzzy) [reply ui/src/app/state/coordinators/chat.ts]: Good catch. Fixed in `e1e06444`: queued Hecate direct-model/task-backed turns now persist `agent_id: "hecate"`, while external-agent queued turns still persist the selected adapter id. I also strengthened the busy-queue UI test to assert the stored queued Hecate turn ownership. Resolving this thread.

**Comment by chicoxyzzy** [comment]
Addressed the latest review feedback in `e1e06444`:

- queued Hecate turns now persist `agent_id: "hecate"` instead of the currently selected external adapter id
- added UI coverage for queued Hecate turn ownership
- added backend coverage for omitted Hecate session titles defaulting to `Hecate Chat`

Verified with:
- `go test ./internal/api ./internal/chat`
- `go vet ./internal/api ./internal/chat`
- `cd ui && bun run format:check`
- `cd ui && bun run typecheck`
- `cd ui && bun run test`

@copilot please review again.
